src/agent/SmartReader.h

```diff
@@ -0,0 +1,13 @@
+#pragma once
+#include <memory>
+#include <string>
+
+class SmartReader
+{
+public:
+
+	SmartReader();
+	std::unique_ptr<SmartReader> GetProperSMARTReader();
```

Comment:
I believe, this should be a method in a SMARTReadersFactory or something similar.